extend.sh

```diff
@@ -245,11 +303,12 @@ function shrink_physical_volume() {
             echo "Failed to resize PV $device during retry: $ret"
             exit $status  
     fi
+    check_filesystem "$LOGICAL_VOLUME_DEVICE_NAME"
 }
 
 function calculate_new_end_partition_size_in_bytes(){
-    local device=$DEVICE_NAME$SUCCESSIVE_PARTITION_NUMBER
-    current_partition_size_in_bytes=$(/usr/sbin/parted -m "$DEVICE_NAME" unit b print| /usr/bin/awk '/^'"$SUCCESSIVE_PARTITION_NUMBER"':/ {split($0,value,":"); print value[3]}'| /usr/bin/sed -e's/B//g')
+    local device=$DEVICE_NAME$1
+    current_partition_size_in_bytes=$(/usr/sbin/parted -m "$DEVICE_NAME" unit b print| /usr/bin/awk '/^'"$1"':/ {split($0,value,":"); print value[3]}'| /usr/bin/sed -e's/B//g')
```

Comment:
Consider assigning `$1` to a local variable to make the code easier to read

Reply:
Done